src/ephemeris.c

```diff
@@ -115,40 +61,24 @@ static void ephemeris_msg_callback(u16 sender_id, u8 len, u8 msg[], void* contex
     return;
   }
 
-  ephemeris_new(&e);
+  u8 v, h;
+  gps_time_t toe;
+  ndb_ephemeris_info(e.sid, &v, &h, &toe, NULL);
+  if (!v || gpsdifftime(&e.toe, &toe) > 0) {
```

Comment:
Yes, makes sense as NDB knows the source. I moved that logic to ndb_ephemeris_store().